tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -1608,76 +1607,65 @@ int main(int argc, char **argv)
+        EXPECT_SUCCESS(s2n_stuffer_skip_read_until(&io_stuffer, (char*) psk->identity.data));
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&io_stuffer, identity_data_size));
         uint32_t obfuscated_ticket_age_1 = 0;
-        EXPECT_SUCCESS(s2n_stuffer_read_uint32(client_io, &obfuscated_ticket_age_1));
+        EXPECT_SUCCESS(s2n_stuffer_read_uint32(&io_stuffer, &obfuscated_ticket_age_1));
 
         /* Skip over the size of the binders list */
-        EXPECT_SUCCESS(s2n_stuffer_skip_read(client_io, sizeof(uint16_t)));
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&io_stuffer, sizeof(uint16_t)));
 
         /* Ensure the binder size is as expected */
         uint8_t binder_1_size = 0;
-        EXPECT_SUCCESS(s2n_stuffer_read_uint8(client_io, &binder_1_size));
+        EXPECT_SUCCESS(s2n_stuffer_read_uint8(&io_stuffer, &binder_1_size));
         EXPECT_TRUE(binder_1_size == BINDER_SIZE);
 
         /* Read the binder from ClientHello 1 */
         uint8_t binder_1[BINDER_SIZE] = { 0 };
-        EXPECT_SUCCESS(s2n_stuffer_read_bytes(client_io, binder_1, BINDER_SIZE));
-        EXPECT_SUCCESS(s2n_stuffer_reread(client_io));
+        EXPECT_SUCCESS(s2n_stuffer_read_bytes(&io_stuffer, binder_1, BINDER_SIZE));
+        EXPECT_SUCCESS(s2n_stuffer_reread(&io_stuffer));
 
         /* Ensure that the ticket age and binder are non-zero */
         EXPECT_TRUE(obfuscated_ticket_age_1 != 0);
         uint8_t zero_array[BINDER_SIZE] = { 0 };
         EXPECT_FALSE(s2n_constant_time_equals(binder_1, zero_array, BINDER_SIZE));
 
-        EXPECT_SUCCESS(s2n_stuffer_copy(client_io, server_io,
-                                        s2n_stuffer_data_available(client_io)));
-
-        /* Server receives ClientHello 1 */
-        EXPECT_SUCCESS(s2n_client_hello_recv(server_conn));
-        EXPECT_SUCCESS(s2n_set_connection_hello_retry_flags(server_conn));
-
-        /* Server sends HelloRetryRequest */
-        EXPECT_SUCCESS(s2n_server_hello_retry_send(server_conn));
-
-        EXPECT_SUCCESS(s2n_stuffer_wipe(client_io));
-        EXPECT_SUCCESS(s2n_stuffer_copy(server_io, client_io,
-                                        s2n_stuffer_data_available(server_io)));
-        client_conn->handshake.message_number = 1; /* hello retry */
-
-        /* Client receives HelloRetryRequest */
-        EXPECT_SUCCESS(s2n_server_hello_recv(client_conn));
-
-        EXPECT_SUCCESS(s2n_stuffer_rewrite(client_io));
+        /* Skip to before the client sends ClientHello 2 */
+        EXPECT_OK(s2n_negotiate_until_message(server_conn, &blocked, HELLO_RETRY_MSG));
+        EXPECT_OK(s2n_negotiate_until_message(server_conn, &blocked, CLIENT_HELLO));
+        EXPECT_OK(s2n_negotiate_until_message(client_conn, &blocked, CLIENT_HELLO));
 
         /* Change the ticket issue time so a new obfuscated ticket age will change */
         psk->ticket_issue_time -= MILLIS_TO_NANOS(1);
 
+        EXPECT_SUCCESS(s2n_stuffer_rewrite(&io_stuffer));
+
```

Comment:
Wouldn't this essentially clear any messages in the stuffer? Won't you lose messages that way? Like, if the client wrote a message, it won't be there for the server to read.

Reply:
Yeah, I put this in when I was hardcoding the stuffer seek values, so I could find the right values more easily. But now that it's seeking relative to identity.data this isn't needed. I removed it in 7c1f62d713858c78bede52a9aea42b9bcdae9703